sql/core/src/main/scala/org/apache/spark/sql/execution/objects.scala

```diff
@@ -389,6 +389,102 @@ case class AppendColumnsWithObjectExec(
     copy(child = newChild)
 }
 
+/**
+ * Groups the input rows together and calls the function with each group and an iterator containing
+ * all elements in the group.  The result of this function is flattened before being output. This
+ * version of the Physical operator takes a user provided initial state.
+ */
+case class MapGroupsWithInitialStateExec(
+    func: (Any, Iterator[Any], LogicalGroupState[Any]) => TraversableOnce[Any],
+    keyDeserializer: Expression,
+    valueDeserializer: Expression,
+    initialStateDeserializer: Expression,
+    groupingAttributes: Seq[Attribute],
+    initialStateGroupingAttributes: Seq[Attribute],
+    dataAttributes: Seq[Attribute],
+    initialStateDataAttrs: Seq[Attribute],
+    outputObjAttr: Attribute,
+    initialState: SparkPlan,
+    timeoutConf: GroupStateTimeout,
+    child: SparkPlan) extends BinaryExecNode with ObjectProducerExec {
+
+  override def left: SparkPlan = child
+
+  override def right: SparkPlan = initialState
+
+  override def outputPartitioning: Partitioning = child.outputPartitioning
```

Comment:
This is something im not sure of

Reply:
what is the output partitioning in MapGroupsExec... it should be the same

Reply:
its the same